kombu/transport/consul.py

```diff
@@ -0,0 +1,272 @@
+    prefix = 'kombu'
+    index = None
+    timeout = '10s'
+    session_ttl = 30
+    lock_name = '%s' % socket.gethostname()
+    queues = {}
+
+    def __init__(self, *args, **kwargs):
+        if consul is None:
+            raise ImportError('Missing python-consul library')
+
+        super(Channel, self).__init__(*args, **kwargs)
+
+        port = self.connection.client.port or self.connection.default_port
+        host = self.connection.client.hostname or DEFAULT_HOST
+
+        LOGGER.debug('Host: %s Port: %d Timeout: %s', host, port, self.timeout)
+
+        self.client = consul.Consul(host=host, port=int(port))
+
+    def _lock_key(self, queue):
+        return '%s/%s.lock' % (self.prefix, queue)
+
+    def _key_prefix(self, queue):
+        return '%s/%s' % (self.prefix, queue)
+
+    def _get_or_create_session(self, queue):
+        """
+        Try to renew the session if it exists, otherwise create a new session
+        in Consul
+
+        This session is used to obtain a lock inside Consul so that we achieve
+        read-consistency between the nodes
+
+        :param queue: The name of the Queue
+        :return: The ID of the session
+        """
+
+        session_id = None
+        try:
+            session_id = self.queues[queue]['session_id']
+        except KeyError:
+            pass
+
+        if session_id is not None:
+            LOGGER.debug('Trying to renew existing session %s',
+                         self.queues[queue]['session_id'])
+            session = self.client.session.renew(session_id=session_id)
+            session_id = None
+            try:
+                session_id = session['ID']
+            except KeyError:
+                pass
+
+        if session_id is None:
+            LOGGER.debug('Creating session %s with TTL %d', self.lock_name,
+                         self.session_ttl)
+            session_id = self.client.session.create(name=self.lock_name,
+                                                    ttl=self.session_ttl)
+
+            LOGGER.debug('Created session %s with id %s', self.lock_name,
+                         session_id)
+
+        return session_id
+
+    def _obtain_lock(self, queue):
+        """
+        Try to obtain a lock on the Queue
+
+        It does so by creating a object called 'lock' which is locked by the
+        current session.
+
+        This way other nodes are not able to write to the lock object which
+        means that they have to wait before the lock is released
+
+        :param queue: The name of the Queue
+        :return: True on success, False otherwise
+        """
+        session_id = self._get_or_create_session(queue)
+        lock_key = self._lock_key(queue)
+
+        LOGGER.debug('Trying to create lock object %s with session %s',
+                     lock_key, session_id)
+
+        if self.client.kv.put(key=lock_key,
+                              acquire=session_id,
+                              value=self.lock_name) is False:
```

Comment:
Do you have a special reason for using `is False`?  I changed it to use `not self.client.kv.put(...)`, as unless there's a comment telling me otherwise, I'm going to expect that this is a mistake :)